service/controller/resource/ipam/azure_machinepool_subnet_checker.go

```diff
@@ -44,20 +46,46 @@ func NewAzureMachinePoolSubnetChecker(config AzureMachinePoolSubnetCheckerConfig
 // AzureMachinePool.
 func (c *AzureMachinePoolSubnetChecker) Check(ctx context.Context, namespace string, name string) (bool, error) {
 	c.logger.LogCtx(ctx, "level", "debug", "message", "checking if node pool subnet has to be allocated")
+	var err error
 
-	objectKey := client.ObjectKey{
-		Namespace: namespace,
-		Name:      name,
+	var azureMachinePool *expcapzv1alpha3.AzureMachinePool
+	{
+		objectKey := client.ObjectKey{
+			Namespace: namespace,
+			Name:      name,
+		}
+
+		azureMachinePool = &expcapzv1alpha3.AzureMachinePool{}
+		err = c.ctrlClient.Get(ctx, objectKey, azureMachinePool)
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+
+		if !azureMachinePool.GetDeletionTimestamp().IsZero() {
```

Comment:
@fiunchinho This is checking the `AzureMachinePool` CR that is currently being reconciled, and bailing out if it is being deleted.